From 45d7199a606c8b1b1ccdcc86270cbec8b9fc62e9 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Wed, 8 Nov 2017 21:58:39 +0000
Subject: [PATCH] cmd/snap-confine: Ensure snap-confine is allowed to access
 os-release

This dupes the existing rule and fixes a regression on Solus where we
see a denial for the os-release file.

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 cmd/snap-confine/snap-confine.apparmor.in | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/cmd/snap-confine/snap-confine.apparmor.in b/cmd/snap-confine/snap-confine.apparmor.in
index 2e1681b7a..88cb344f7 100644
--- a/cmd/snap-confine/snap-confine.apparmor.in
+++ b/cmd/snap-confine/snap-confine.apparmor.in
@@ -213,6 +213,9 @@
     umount /var/lib/snapd/hostfs/proc/,
     mount options=(rw rslave) -> /var/lib/snapd/hostfs/,
 
+    # Allow reading the os-release file (possibly a symlink to /usr/lib).
+    /{etc/,usr/lib/}os-release r,
+
     # set up snap-specific private /tmp dir
     capability chown,
     /tmp/ w,
-- 
2.15.0

